lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,318 @@
+  const adjustBtnFocus = (adjust: (i: number) => number) => () => {
+        const newFocusableBtnIndex = adjust(focusableBtnIndex ?? 0),
+            elToFocus = dropdownRef.current?.children[newFocusableBtnIndex] as HTMLElement | null;
+
+        if (elToFocus) {
+          setFocusableBtnIndex(newFocusableBtnIndex);
+          elToFocus.scrollIntoView({ block: 'nearest' });
+
+          if (autoComplete) {
+            setInlineStyle(false);
+          }
+        }
+      },
+      focusNext = adjustBtnFocus(inc),
+      focusPrev = adjustBtnFocus(dec),
+      focusFirst = adjustBtnFocus(always(0)),
+      focusLast = adjustBtnFocus(always(matches.length - 1));
+
+  function handleOnClick({displayName}: DataItem<T, string>) {
+    onChange(displayName);
+    onSearch(displayName);
+    focusTextInput();
+    setFocusableBtnIndex(null);
+    setShowDropdown(false);
+  }
+
+  function handleKeyDown(evt: KeyboardEvent<HTMLElement>) {
+    const inputEle = evt.currentTarget as HTMLInputElement,
+        elToFocusText = focusableBtnIndex !== null ? matches[focusableBtnIndex].displayName : null;
+    switch (evt.key) {
+      case 'Enter':
+        if (focusableBtnIndex !== null) {
+          handleOnClick(matches[focusableBtnIndex]);
+        }
+        evt.preventDefault();
+        break;
+      case 'Home':
+        inputEle.setSelectionRange(0, 0);
+        evt.preventDefault();
+        break;
+      case 'End':
+        autoComplete && elToFocusText
+          ? inputEle.setSelectionRange(elToFocusText.length, elToFocusText.length)
+          : inputEle.setSelectionRange(value.length, value.length);
+        evt.preventDefault();
+        break;
+      case 'ArrowDown':
+        if (!isEmpty) {
+          setShowDropdown(true);
+        }
+
+        focusableBtnIndex !== null
+          ? (focusableBtnIndex === matches.length - 1 ? focusFirst() : focusNext())
+          : focusFirst();
+
+        evt.preventDefault();
+        break;
+      case 'ArrowUp':
+        if (!isEmpty) {
+          setShowDropdown(true);
+        }
+
+        focusableBtnIndex !== null
+          ? (focusableBtnIndex === 0 ? focusLast() : focusPrev())
+          : focusLast();
+
+        evt.preventDefault();
+        break;
+      case 'Backspace':
+        if (autoComplete) {
+          setInlineStyle(false);
+        }
+        break;
+      case 'Escape':
+        if (showDropdown) {
+          setShowDropdown(false);
+        }
+        else {
+          onChange('');
+          setFocusableBtnIndex(null);
+        }
+        break;
+      default:
+        if (autoComplete) {
+          setInlineStyle(true);
+        }
+        break;
+    }
+  }
+
+  useEffect(function() {
+    // Nullify focusableBtnIndex whenever the number of matches changes
+    setFocusableBtnIndex(null);
+
+    if (matches.length && autoComplete && showDropdown && inlineStyle) {
+      // If the typed characters match the beginning of the name of option in the dropdown,
```

Comment:
I think for proper functioning of the component (see https://github.com/sonatype/sonatype-react-shared-components/pull/751#discussion_r964844937) we have to offload this assumption to the calling code.  That is, we need to add to the documentation that the search logic _must_ only return items that start with the search text.  Once we are operating under that assumption, we can remove the condition below as it should always be true.

Reply:
Updated here 7047f0760fdf092735b418b118ce8b2643a1d17c.